include/algorithms/synthesis/syrec_synthesis.hpp

```diff
@@ -0,0 +1,160 @@
+        virtual bool less_than(unsigned dest, const std::vector<unsigned>& src1, const std::vector<unsigned>& src2);                          // <
+        virtual bool modulo(const std::vector<unsigned>& dest, const std::vector<unsigned>& src1, const std::vector<unsigned>& src2);         // %
+        virtual bool multiplication(const std::vector<unsigned>& dest, const std::vector<unsigned>& src1, const std::vector<unsigned>& src2); // *
+        virtual bool not_equals(unsigned dest, const std::vector<unsigned>& src1, const std::vector<unsigned>& src2);                         // !=
+        virtual bool swap(const std::vector<unsigned>& dest1, const std::vector<unsigned>& dest2);                                            // <=>
+        static bool  check_repeats();
+
+        bool decrease_new(const std::vector<unsigned>& rhs, const std::vector<unsigned>& lhs);
+        bool decrease_new_assign(const std::vector<unsigned>& rhs, const std::vector<unsigned>& lhs);
+        bool increase_new(const std::vector<unsigned>& rhs, const std::vector<unsigned>& lhs);
+        bool expression_op_inverse(unsigned op, const std::vector<unsigned>& exp_lhs, const std::vector<unsigned>& exp_rhs);
+        bool expression_single_op(unsigned op, const std::vector<unsigned>& exp_lhs, const std::vector<unsigned>& exp_rhs);
+        bool exp_evaluate(std::vector<unsigned>& lines, unsigned op, const std::vector<unsigned>& lhs, const std::vector<unsigned>& rhs);
+        // shift operations
+        virtual bool left_shift(const std::vector<unsigned>& dest, const std::vector<unsigned>& src1, unsigned src2);  // <<
+        virtual bool right_shift(const std::vector<unsigned>& dest, const std::vector<unsigned>& src1, unsigned src2); // >>
+
+        // efficient controls
+        bool add_active_control(unsigned);
+        bool remove_active_control(unsigned);
+
+        virtual bool assemble_circuit(const cct_node&);
+
+        cct_manager cct_man;
+
+        void         add_variable(circuit& circ, const std::vector<unsigned>& dimensions, const applications::variable::ptr& var, constant _constant, bool _garbage, const std::string& arraystr);
+        virtual bool get_variables(const applications::variable_access::ptr& var, std::vector<unsigned>& lines);
+        virtual bool unget_variables(const applications::variable_access::ptr& var, std::vector<unsigned>& lines);
+
+        unsigned get_constant_line(bool value);
+        bool     get_constant_lines(unsigned bitwidth, unsigned value, std::vector<unsigned>& lines);
+
+    private:
+        circuit&                                    _circ;
+        std::stack<applications::statement::ptr>    _stmts;
+        var_lines_map                               _var_lines;
+        std::map<bool, std::vector<unsigned>>       free_const_lines_map;
+        applications::number::loop_variable_mapping loop_map;
+
+        std::stack<applications::module::ptr> modules;
+
+        std::string variable_name_format;
```

Comment:
Is this variable name format supposed to change anywhere?
As far as I can tell it is just set to its default when constructing the synthesizer and never really modified or used in a different fashion. Are there any use cases I am missing where changing this format would help?
If not, then I would argue that this can be safely removed. This also allows to drop the `boost/format` include by replacing the single occurrence with C++ string concatenation.